tests/common/snappi/snappi_fixtures.py

```diff
@@ -517,3 +521,361 @@ def cvg_api(snappi_api_serv_ip,
+        dutIp = dutIps[port_id]
+        tgenIp = tgenIps[port_id]
+        mac = __gen_mac(port_id)
+        logger.info('Configuring Dut: {} with port {} with IP {}/{}'.format(
+                                                                            duthost.hostname,
+                                                                            port['peer_port'],
+                                                                            dutIp,
+                                                                            prefix))
+        if port['asic_value'] == 'None':
+            duthost.command('sudo config interface ip add {} {}/{} \n' .format(
+                                                                                port['peer_port'],
+                                                                                dutIp,
+                                                                                prefix))
+        else:
+            duthost.command('sudo config interface -n {} ip add {} {}/{} \n' .format(
+                                                                                    port['asic_value'],
+                                                                                    port['peer_port'],
+                                                                                    dutIp,
+                                                                                    prefix))
+        device = config.devices.device(name='Device Port {}'.format(port_id))[-1]
+        ethernet = device.ethernets.add()
+        ethernet.name = 'Ethernet Port {}'.format(port_id)
+        ethernet.port_name = config.ports[port_id].name
+        ethernet.mac = mac
+        ip_stack = ethernet.ipv4_addresses.add()
+        ip_stack.name = 'Ipv4 Port {}'.format(port_id)
+        ip_stack.address = tgenIp
+        ip_stack.prefix = prefix
+        ip_stack.gateway = dutIp
+        port_config = SnappiPortConfig(
+                                        id=port_id,
+                                        ip=tgenIp,
+                                        mac=mac,
+                                        gw=dutIp,
+                                        gw_mac=str(duthost.facts['router_mac']),
+                                        prefix_len=prefix,
+                                        port_type=SnappiPortType.IPInterface,
+                                        peer_port=port['peer_port']
+                                      )
+        port_config_list.append(port_config)
+
+    return True
+
+
+def get_dut_interconnected_ports(conn_graph_facts, src_host, dst_host):     # noqa: F811
+    ports = []
+    for key, value in conn_graph_facts['device_conn'][src_host].items():
+        if value['peerdevice'] == dst_host:
+            ports.append((key, value['peerport']))
+    return ports
+
+
+def get_multidut_tgen_peer_port_set(line_card_choice, ports, config_set, number_of_tgen_peer_ports=2):
```

Comment:
Why tgen?

Reply:
this is just a naming that has been used, no specific reason to it